### DISCUSSION TITLE
Overlaying gif animation on top of static image

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#2564] [Help] [answered]
I have two `.gif` images, one animated (a weather radar display) and the other the static background (map), so I want the map to be the background of the final animation. How do I accomplish this?

**Comment by fmw42** [comment]
What is your IM version and platform/OS? Syntax may vary.

For IM 6

`convert background.gif  null:  image.gif -coalesce -compose over -layers composite -layers optimize result.gif`

See https://imagemagick.org/Usage/anim_mods/#composite

**Reply by ghost** [reply]
- The imagemagick.org link at the end of the message is a 404.

- `ImageMagick 6.9.10-23 Q16 x86_64 20190101` on Ubuntu 20.04.1

- The command line you proposed returned the following error:

```
> convert radarback.gif null: radar.gif -coalesce -compose over -layers composite -layers optimize result.gif
convert-im6.q16: missing Null Image List Separator layers Composite @ error/mogrify.c/MogrifyImageList/8454.
```
The other idea I have is to compose the individual frames and then combine them back together. I'll try that.

**Comment by fmw42** [comment]
`convert image.gif -coalesce null: background.gif -compose dst_over -layers composite -layers optimize result.gif`

**Comment by ghost** [comment, marked as answer]
I figured it out on my own. I just used successive `-page` options and then `-flatten`ed the result into each frame and then used `ffmpeg` to compile the animation frames into a video (because using glob in `convert` maxed out the command line). The individual frames are named according to the UNIX epoch seconds that they were downloaded.

```sh
#!/bin/sh
for f in 1*.gif
  do convert -compose src-over -page 0,0 TBW_Topo_Short.jpg -page 0,0 $f -page 0,0 TBW_County_Short.gif -page 0,0 TBW_Highways_Short.gif -page 0,0 TBW_City_Short.gif -page 0,0 label:"$(date --date="@$(basename $f .gif)")" -flatten $(basename $f .gif).png
done
ffmpeg -r:v 30 -pattern_type glob -i '*.png' -codec:v libx264 -preset veryslow -pix_fmt yuv420p -crf 28 -an "radar.mp4"
rm *.png
```

Result: <https://www.instagram.com/p/CFFopZgFDSH/>
